### PR TITLE
[DNM] Update readme document

### PR DESCRIPTION
**Pull request by FelixYBW**
Update readme document, make it more clear

## What changes were proposed in this pull request?
Only document

## How was this patch tested?
NA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,107 +1,115 @@
...
+* For unsupported operators or functions Gluten fallback the operator to Vanilla Spark. There are C2R and R2C converter to convert the columnar data and Spark's internal row data. Both C2R and R2C are implemented natively as well
+* Metrics are very important to get insight of Spark's execution, identify the issues or bottlenecks. Gluten collects the metrics from native library and shows in Spark UI.
+* Shim layer is used to support mutiple releases of Spark. Gluten does not and will not support all the Spark releases. The plan is to support the latest 2-3 spark stable release only.
 
-Not all data type is supported now. Refere to Velox and Clickhouse to check the supported data types.
+# Usage
 
-Operator stat info is pretty useful to understand Spark's execution status. Velox metrics are already added to Spark metric list for Velox backend. More info about Velox's stat can be found here: https://facebookincubator.github.io/velox/develop/debugging/print-plan-with-stats.html
+Gluten is still under actively developping now. There isn't released binary yet. The only way to use Gluten is to build from source.
```
Comment (akashsha1) [inline README.md:45]: nit. fixing some sentences: 
Gluten is still under active development now. There isn't a released binary yet. The only way to use Gluten is to build from source.

File: README.md [hunk 2]
```diff
@@ -1,107 +1,115 @@
...
+# Contribution
 
-### Build Gluten with Arrow backend
+Gluten project welcomes everyone to contribute. 
 
-If you would like to build Gluten with **Arrow** backend, please follow the steps in [Build with Arrow Backend](./docs/ArrowBackend.md).
+## Community
 
-### Build Gluten with ClickHouse backend
+Currently we communite with all developers and users in a wechat group(Chinese only), Spark channel in Velox Slack group. Contact us if you would like to join in.
```
Comment (akashsha1) [inline README.md:74]: nit. communicate

File: README.md [hunk 3]
```diff
@@ -1,107 +1,115 @@
...
+## Bug Reports
 
-If you would like to build Gluten with **ClickHouse** backend, please follow the steps in [Build with ClickHouse Backend](./docs/ClickHouse.md). ClickHouse backend is devleoped by [Kyligence](https://kyligence.io/), please visit https://github.com/Kyligence/ClickHouse for more infomation.
+Feel free to submit any bugs, issues or enhancement requirements to github issue list. Be sure to follow the bug fill template so we can solve it quickly. If you already implement a PR and would like to contribute, you may submit an issue firstly and refer to the issue in the PR. 
 
-### Build script parameters
+## Documentation
 
-[Gluten Usage](./docs/GlutenUsage.md) listed the parameters and their default value of build command for your reference
+Unfortunately we haven't create the orginazed documentation site for Gluten. Currently all document is hold in [docs](https://github.com/oap-project/gluten/tree/main/docs). Ping us if you would like to know more details about the Gluten design. Gluten is still in developping now, some designs may change. Feel free to talk with us if you have better design.
```
Comment (akashsha1) [inline README.md:82]: Unfortunately we haven't organized the documentation site for Gluten.

Comment (akashsha1) [inline README.md:82]: Gluten is still under development now, and some designs may change. Feel free to talk with us and share other design and ideas.

File: README.md [hunk 4]
```diff
@@ -1,107 +1,115 @@
...
 
-The testing environment is using single node with 2TB datasize and using Spark3.1.1 for both baseline and Gluten. The Decision Support Benchmark1 result shows an average speedup of 2.07x and up to 8.1X speedup in a single query with Gluten and Velox backend. Spark3.2 performance is pretty close.
+The testing environment is using single node with 2TB datasize and using Spark3.1.1 for both baseline and Gluten. The Decision Support Benchmark1 result shows an average speedup of 2.07x and up to 8.1X speedup in a single query with Gluten and Velox backend. Spark3.2 performance is pretty close. Performance data is tested in Sep. 2022. Contact us if you'd like to know latest performance number
```
Comment (akashsha1) [inline README.md:89]: Contact us if you'd like to know about the latest performance results. 

Also suggest adding some way any customer can contact you - email id, slack handle, post questions via issue tabs etc.

File: README.md [hunk 5]
```diff
@@ -1,107 +1,115 @@
...
+<p align="center">
+<img src="https://user-images.githubusercontent.com/47296334/199617207-1140698a-4d53-462d-9bc7-303d14be060b.png" width="800">
+</p>
+There are several native libraries we may offload. Currently we are working on Clickhouse and Velox as native backend. Velox is a C++ database acceleration library which provides reusable, extensible, and high-performance data processing components. More details can be found from https://github.com/facebookincubator/velox/. We also implemented a basic backend using Arrow Computer Engine which is for reference only. Gluten can also be easily extended to any accelerator libraries as backend.
 
-Not all the operators and functions are added. 22 queries of TPCH can pass. We are working on TPCDS queries. <b> We haven't pass Spark UTs, the work is just started and expected to finish in Q4 2022. So any other queries except H/DS may run into issues. Feel free to submit it to our issue list.</b>
+There are several key component in GLuten:
```
Comment (akashsha1) [inline README.md:35]: nit. capitalization: 
Gluten

File: README.md [hunk 6]
```diff
@@ -1,107 +1,115 @@
...
+* Query plan conversion which convert Spark's physical plan into substrait plan in each stage.
+* Unified memory management in Spark is used to control the native memory allocation as well
+* Columnar shuffle is used to shuffle columnar data directly. The shuffle service still reuse the one in Spark core. The exchange operator is reimplemented to support columnar data format
+* For unsupported operators or functions Gluten fallback the operator to Vanilla Spark. There are C2R and R2C converter to convert the columnar data and Spark's internal row data. Both C2R and R2C are implemented natively as well
+* Metrics are very important to get insight of Spark's execution, identify the issues or bottlenecks. Gluten collects the metrics from native library and shows in Spark UI.
+* Shim layer is used to support mutiple releases of Spark. Gluten does not and will not support all the Spark releases. The plan is to support the latest 2-3 spark stable release only.
```
Comment (akashsha1) [inline README.md:41]: Gluten does not and will not support all the Spark releases. The plan is to support the latest 2-3 spark stable release only.

>> Gluten only plans to support the latest 2-3 spark stable releases, with no plans to add support on older spark releases. Current support is on Spark 3.2 and Spark 3.3.

File: README.md [hunk 7]
```diff
@@ -1,107 +1,115 @@
...
+Apache Spark is a stable, mature project that has been under development for many years. The project has proven to be one of the best frameworks to scale out of processing petabyte-scale datasets. However, the Spark community has had to address performance challenges that required various optimizations over time. A key optimization introduced in Spark 2.0 replaced Volcano mode with whole-stage code-generation to achieve a 2x speedup. Since then most of the optimization works at the query plan level. The operator's performance stopped to grow.
 
-![Overview](./docs/image/overall_design.png)
+<p align="center">
+<img src="https://user-images.githubusercontent.com/47296334/199614366-6209ed91-8955-4c7a-8829-e80ccdfaded9.png" width="800">
+</p>
```
Comment (akashsha1) [inline README.md:10]: image needs a title describing what it is showing.

File: README.md [hunk 8]
```diff
@@ -1,107 +1,115 @@
...
+* Manage data sharing between JVM and native
+* Extend support to native accelerators
 
-# Issues to Solve and status
+## Architecture
 
-The code is still not completely cleaned now. The work is still WIP.
+The overview chart is like below. Spark physical plan is transformed to substrait plan. Substrait is to create a well defined cross-language specification for data compute operations. More details can be found from https://substrait.io/. Then substrait plan is passed to native through JNI call. In native the operator chain should be built and start to run. We use Spark3.0's columnar API as the data interface, so the native library should return Columnar Batch to Spark. We may need to wrap the columnar batch for each native backend. Gazelle engine's c++ code use Apache Arrow data format as its basic data format, so the returned data to Spark JVM is ArrowColumnarBatch.
+<p align="center">
+<img src="https://user-images.githubusercontent.com/47296334/199617207-1140698a-4d53-462d-9bc7-303d14be060b.png" width="800">
```
Comment (akashsha1) [inline README.md:31]: image needs title describing what it is showing.

File: README.md (hunk 7 above)
Comment (FelixYBW) [reply README.md:10]: Done

File: README.md (hunk 4 above)
Comment (FelixYBW) [reply README.md:89]: Contact info is at the end of doc

**Review by zzcclp — APPROVED** [review]
LGTM

**Comment by zzcclp** [comment]
Please rebase

File: README.md [hunk 9]
```diff
@@ -1,107 +1,131 @@
...
+## 1.3 Target User
 
-![Overview](./docs/image/operators.png)
+Gluten targets to the Spark administrators and Spark users who want to improve their spark cluster's performance fundamentally. Gluten is an plunin to Spark. It's designed to offload the SQL engine to native without any dataframe API or SQL query changes. SparkSQL users can run their current Spark job on Gluten seemlessly, no code changes are needed. However as an plugin, Gluten needs some configurations to enable it when you start Spark context. All configurations are listed [here](https://github.com/oap-project/gluten/blob/main/docs/Configuration.md)
```
Comment (carsonwang) [inline README.md:29]: nit: (1) Gluten is `an plunin` to Spark -> a plugin  (2) However as `an` plugin -> a plugin.

File: README.md [hunk 10]
```diff
@@ -1,107 +1,131 @@
...
 
-![Overview](./docs/image/flow.png)
+There are several key component in Gluten:
+* Query plan conversion which convert Spark's physical plan into substrait plan in each stage.
+* Unified memory management in Spark is used to control the native memory allocation as well
+* Columnar shuffle is used to shuffle columnar data directly. The shuffle service still reuse the one in Spark core. The exchange operator is reimplemented to support columnar data format
+* For unsupported operators or functions Gluten fallback the operator to Vanilla Spark. There are C2R and R2C converter to convert the columnar data and Spark's internal row data. Both C2R and R2C are implemented natively as well
+* Metrics are very important to get insight of Spark's execution, identify the issues or bottlenecks. Gluten collects the metrics from native library and shows in Spark UI.
+* Shim layer is used to support mutiple releases of Spark. Gluten only plans to support the latest 2-3 spark stable releases, with no plans to add support on older spark releases. Current support is on Spark 3.2 and Spark 3.3.
```
Comment (carsonwang) [inline README.md:54]: nit: mutiple  -> multiple

File: README.md (hunk 9 above)
Comment (carsonwang) [inline README.md:29]: nit: seemlessly -> seamlessly

File: README.md [hunk 11]
```diff
@@ -1,107 +1,131 @@
...
+<p align="center">
+<img src="https://user-images.githubusercontent.com/47296334/199617207-1140698a-4d53-462d-9bc7-303d14be060b.png" width="800">
+</p>
+There are several native libraries we may offload. Currently we are working on Clickhouse and Velox as native backend. Velox is a C++ database acceleration library which provides reusable, extensible, and high-performance data processing components. More details can be found from https://github.com/facebookincubator/velox/. We also implemented a basic backend using Arrow Computer Engine which is for reference only. Gluten can also be easily extended to any accelerator libraries as backend.
 
-![Overview](./docs/image/flow.png)
+There are several key component in Gluten:
+* Query plan conversion which convert Spark's physical plan into substrait plan in each stage.
+* Unified memory management in Spark is used to control the native memory allocation as well
+* Columnar shuffle is used to shuffle columnar data directly. The shuffle service still reuse the one in Spark core. The exchange operator is reimplemented to support columnar data format
```
Comment (carsonwang) [inline README.md:51]: nit: reuse -> reuses

File: README.md [hunk 12]
```diff
@@ -1,107 +1,131 @@
...
+* For unsupported operators or functions Gluten fallback the operator to Vanilla Spark. There are C2R and R2C converter to convert the columnar data and Spark's internal row data. Both C2R and R2C are implemented natively as well
+* Metrics are very important to get insight of Spark's execution, identify the issues or bottlenecks. Gluten collects the metrics from native library and shows in Spark UI.
+* Shim layer is used to support mutiple releases of Spark. Gluten only plans to support the latest 2-3 spark stable releases, with no plans to add support on older spark releases. Current support is on Spark 3.2 and Spark 3.3.
 
-# Issues to Solve and status
+# 3 Usage
 
-The code is still not completely cleaned now. The work is still WIP.
+Gluten is still under active development now. There isn't a released binary yet. The only way to use Gluten is to build from source, copied the jar to your spark jars, then enable Gluten plugin when you start your spark context. Here is the simple example. Refer to Velox or Clickhouse backend below for more details
```
Comment (carsonwang) [inline README.md:58]: nit: copied -> copy

File: README.md [hunk 13]
```diff
@@ -1,107 +1,131 @@
...
 
-Spark's RDD cache is columnar batch based. In Gazelle-plugin, we use the arrow record batch directly without any memcpy. We can build the same functionality in Gluten.
+## 3.2 Build and Install Gluten with ClickHouse backend
 
-Pyspark support needs to be ported as well. If input data format is Arrow, we can send the data to pyspark directly. No memcpy
+![logo](./docs/image/ClickHouse/logo.png)
 
-UDF support. We need to create the interface which use columnar batch as input. So customer can port their current UDF to columnar batch based. If it's Arrow record batch based, user can utilize Arrow's JAVA or C++ API to implement their UDF, debug without Spark, then register to Spark.
+If you would like to build and try  Gluten with **ClickHouse** backend, please follow the steps in [Build with ClickHouse Backend](./docs/ClickHouse.md). ClickHouse backend is devleoped by [Kyligence](https://kyligence.io/), please visit https://github.com/Kyligence/ClickHouse for more infomation.
```
Comment (carsonwang) [inline README.md:81]: nit: devleoped -> developed

File: README.md [hunk 14]
```diff
@@ -1,107 +1,131 @@
...
+Apache Spark is a stable, mature project that has been under development for many years. The project has proven to be one of the best frameworks to scale out of processing petabyte-scale datasets. However, the Spark community has had to address performance challenges that required various optimizations over time. A key optimization introduced in Spark 2.0 replaced Volcano mode with whole-stage code-generation to achieve a 2x speedup. Since then most of the optimization works at the query plan level. The operator's performance stopped to grow.
 
-##### Before we enable Gazelle's C++ code as backend, we will continue Gazelle's development.
+<p align="center">
+<img src="https://user-images.githubusercontent.com/47296334/199853029-b6d0ea19-f8e4-4f62-9562-2838f7f159a7.png" width="800">
+</p>
 
-![Overview](./docs/image/gluten.png)
+On the other side, SQL engine is researched for years. There are product or libraries like Clickhouse, Arrow or Velox. By using features like native implementation, columnar data format as well as vectorized data processing, these libraries outperform much of Spark's JVM based SQL eingine. However these libraries are running on single node.
```
Comment (carsonwang) [inline README.md:12]: nit: eingine -> engine

**Review by carsonwang — COMMENTED** [review]
Looks good! Much clear now. Just some minor typos to fix.

File: README.md [hunk 15]
```diff
@@ -1,107 +1,131 @@
...
 
-# Reference
 
-Please check below links for more related information.
-- [Gluten Intro Video at Data AI Summit 2022](https://databricks.com/dataaisummit/session/gazelle-jni-middle-layer-offload-spark-sql-native-engines-execution)
-- [Gluten Intro Article at Medium.com](https://medium.com/intel-analytics-software/accelerate-spark-sql-queries-with-gluten-9000b65d1b4e)
-- [Gluten Intro Article at Kyligence.io(in Chinese)](https://cn.kyligence.io/blog/gluten-spark/)
 
-# Contact
+# 5 License
```
Comment (carsonwang) [inline README.md:120]: nit: 6 License

**Pull request merged**